rs/index/src/ivf/builder.rs

```diff
@@ -263,14 +273,37 @@ impl<D: DistanceCalculator + CalculateSquared + Send + Sync> IvfBuilder<D> {
         Ok(distances)
     }
 
+    fn get_num_valid_vectors(&self) -> usize {
+        self.valid_point_id_mapping.len()
+    }
+
+    fn is_valid_point_id(&self, point_id: usize) -> bool {
```

Comment:
a faster way to do this is (since the usecase for this is to get all valid point_id), is to get the values of `valid_point_id_mapping`. That way you don't need so many map lookups